### DISCUSSION TITLE
Cppcheck Linux

### DISCUSSION THREAD

**Discussion by AAsyunkin-se** [github0null/eide#229] [Q&A] [answered]
Hi all,

Is there a Wiki or a doc that shows how to setup Cppcheck (Debian Linux)?

I installed cppcheck via apt, modified eide config to point to cppcheck at /usr/bin/cppcheck, but when trying to run cppcheck via Ctrl-Alt-C I get:

`cppcheck: No C or C++ source files found.`

This happens with both 'Multi' and simple projects.. What am I missing?

**Comment by github0null** [comment]
Sorry I'm late

Before call the cppcheck executable, the plug-in will genrate a configuration for cppcheck:

![image](https://github.com/github0null/eide/assets/43022536/9bf6728f-0aef-4fa8-beb1-b028fca35b68)

All of your project informations will be in this file, you can check it.

**Reply by AAsyunkin-se** [reply]
Yes, that's correct. The cppcheck config appears to be at the correct place and populated with correct filenames of my project files. However, as I described above, trying to run cppcheck via Ctrl-Alt-C doesn't work. Is there a way to get debug logs of the plugin?

**Reply by github0null** [reply]
I'm not sure what happened.

This plug-in use these command to launch cppcheck:

```shell
cppcheck -j 4 \
    --error-exitcode=0 \
    --report-progress \
    --enable=warning --enable=performance --enable=portability \
    --project=<tmp.cppcheck FILE PATH> \
    --relative-paths=<PROJECT_ROOT_DIR>
```

you can try it in linux bash

**Reply by AAsyunkin-se** [reply, marked as answer]
Ok, I think I got to the bottom of this.

Ubuntu repos install ridiculously old cppcheck versions. On 1804 it installed 1.84, dating many years back. So features like --project were not even supported. That's the reason I saw 'No C or C++ source files found'.
On 2004 it installs 1.96 or something...where the latest is 2.10. Also not great.

You are probably on Windows - the only target that cppcheck creators maintain. :)

Solution for non-Win - download latest package, build it and install via 'sudo make install'. Installation is mandatory as otherwise cppcheck won't find files it needs to run the check (in my case it complained that platform arm32-w2t doesn't exists).

Thank you for the hint on launching cppcheck manually - this made my life easier!
You might want to add this to a wiki or something - I am sure there are a lot of GCC/Linux users of VS Code here..

Cheers.

**Reply by github0null** [reply]
> You might want to add this to a wiki or something - I am sure there are a lot of GCC/Linux users of VS Code here..

OK, I will add it to the document later
